src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -344,6 +344,30 @@ public function setAnnotatedClassCache(array $annotatedClasses)
     }
 
     /**
+     * Return a list of classes to be cached away when the container is build.
+     *
+     * If a class contains annotations, add it to getAnnotatedClassesToCompile()instead.
+     *
+     * @return string[]
+     */
+    public function getClassesToCompile()
```

Comment:
I'm planning to verify this soon. Yet this PR is right adding it: it's just moving existing things around, and we still support php 5.
If php7 makes this unnecessary, then the whole inline system could be deprecated.
To be proven first, and not in this PR imho.